File: Cryptography/atbash.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

#define MAX_TEXT_SIZE 1000

void atbashCipher() {
    char mode;
    char text[MAX_TEXT_SIZE];
    
    // Get mode (encrypt or decrypt)
    printf("Do you want to (e)ncrypt or (d)ecrypt? ");
    scanf("%c", &mode);
    getchar(); // Clear the newline character
    
    // Get text
    printf("Enter the text: ");
    fgets(text, MAX_TEXT_SIZE, stdin);
    text[strcspn(text, "\n")] = '\0'; // Remove trailing newline
    
    // Process the text (Atbash is its own inverse, so encryption and decryption are the same)
    for (int i = 0; i < strlen(text); i++) {
        if (isalpha(text[i])) {
            if (islower(text[i])) {
                text[i] = 'z' - (text[i] - 'a');
            } else {
                text[i] = 'Z' - (text[i] - 'A');
            }
        }
    }
    
    printf("Result: %s\n", text);
}

int main() {
    atbashCipher();
    return 0;
}

File: atbash.py
def atbash_cipher(text, mode):
    result = ""
    for char in text:
        if char.isalpha():
            # Determine if uppercase or lowercase
            if char.isupper():
                # For uppercase: Z - (char - A)
                result += chr(90 - (ord(char) - 65))
            else:
                # For lowercase: z - (char - a)
                result += chr(122 - (ord(char) - 97))
        else:
            # Keep non-alphabetic characters unchanged
            result += char
    
    return result

def main():
    # Get user input
    message = input("Enter the message: ")
    
    while True:
        mode = input("Enter 'encrypt' for encryption or 'decrypt' for decryption: ").lower()
        if mode in ["encrypt", "decrypt"]:
            break
        else:
            print("Please enter either 'encrypt' or 'decrypt'.")
    
    # Process the message (Atbash is its own inverse, so encrypt/decrypt are the same)
    result = atbash_cipher(message, mode)
    
    # Display the result
    print(f"\nOriginal message: {message}")
    print(f"Mode: {mode}")
    print(f"Result: {result}")

if __name__ == "__main__":
    main()
